src/test/java/edu/cornell/kfs/concur/batch/service/impl/MockConcurRequestV4ServiceEndpoint.java

```diff
@@ -0,0 +1,222 @@
+
+    private static final String REQUESTS_ENDPOINT_HANDLER_PATTERN = "/travelrequest/v4/*";
+
+    private static final Pattern REQUESTS_ENDPOINT_REGEX = Pattern.compile(
+            "^/travelrequest/v4/requests("
+                    + "(?<getRequestList>\\?(([\\w%\\-\\.:]+=[\\w%\\-\\.:]*)(\\&[\\w%\\-\\.:]+=[\\w%\\-\\.:]*)*))|"
+                    + "(?<getRequest>/(?<requestUuid>\\w+)/?)"
+            + ")$");
+
+    private static final String GET_REQUEST_LIST_GROUP = "getRequestList";
+    private static final String GET_REQUEST_GROUP = "getRequest";
+    private static final String REQUEST_UUID_GROUP = "requestUuid";
+
+    private static final String ERRORS_PROPERTY = "errors";
+    private static final String ERROR_CODE_PROPERTY = "errorCode";
+    private static final String ERROR_MESSAGE_PROPERTY = "errorMessage";
+
+    private static final String TEST_ERROR_CODE = "testError";
+    private static final String NOT_FOUND_MESSAGE = "Not Found";
+    private static final String INTERNAL_SERVER_ERROR_MESSAGE = "Internal Server Error";
+
+    private ObjectMapper objectMapper;
+    private MockConcurRequestV4Backend mockBackendServer;
+    private String expectedAccessToken;
+    private RequestV4DetailFixture[] initialRequestDetails;
+    private String baseRequestV4Url;
+    private boolean forceServerError;
+
+    public MockConcurRequestV4ServiceEndpoint(String expectedAccessToken,
+            RequestV4DetailFixture... initialRequestDetails) {
+        this.objectMapper = buildObjectMapper();
+        this.mockBackendServer = null;
+        this.expectedAccessToken = expectedAccessToken;
+        this.initialRequestDetails = initialRequestDetails;
+        this.forceServerError = false;
+    }
+
+    private ObjectMapper buildObjectMapper() {
+        ObjectMapper jsonObjectMapper = new ObjectMapper();
+        jsonObjectMapper.setTimeZone(TimeZone.getTimeZone(CUKFSConstants.TIME_ZONE_UTC));
+        jsonObjectMapper.setDateFormat(new SimpleDateFormat(
+                CUKFSConstants.DATE_FORMAT_yyyy_MM_dd_T_HH_mm_ss_SSS_Z, Locale.US));
+        return jsonObjectMapper;
+    }
+
+    public void setForceServerError(boolean forceServerError) {
+        this.forceServerError = forceServerError;
+    }
+
+    @Override
+    public String getRelativeUrlPatternForHandlerRegistration() {
+        return REQUESTS_ENDPOINT_HANDLER_PATTERN;
+    }
+
+    @Override
+    protected void onServerInitialized(String baseUrl) {
+        super.onServerInitialized(baseUrl);
+        baseRequestV4Url = baseUrl + ParameterTestValues.REQUEST_V4_RELATIVE_ENDPOINT;
+        mockBackendServer = new MockConcurRequestV4Backend(baseRequestV4Url, initialRequestDetails);
+    }
+
+    public MockConcurRequestV4Backend getMockBackendServer() {
+        if (ObjectUtils.isNull(mockBackendServer)) {
+            throw new IllegalStateException("Mock back-end server was null; this endpoint may not "
```

Comment:
similarly, can some of this error handling code be removed to simplify this mock service?

Reply:
In the case of this getMockBackendServer() method, it's not actively used by the tests and won't be needed until we implement the workflow travel request handling in a future user story. I'll go ahead and remove this getter for now.

As for the getBaseRequestV4Url() method, it's extremely important that it returns the properly initialized URL; otherwise the calling code in the unit test class won't be able to invoke the local endpoint. I would prefer to keep that one in place for now.

Reply:
OK, thanks for explaining